src/tests/ftest/launch.py

```diff
@@ -154,70 +160,81 @@ def set_test_environment(args):
-        net_path = os.path.join(os.path.sep, "sys", "class", "net")
-        net_list = [dev for dev in os.listdir(net_path) if dev != "lo"]
-        for device in sorted(net_list):
-            # Get the interface state - only include active (up) interfaces
-            with open(os.path.join(net_path, device, "operstate"), "r") as \
-                 fileh:
-                state = fileh.read().strip()
-            # Only include interfaces that are up
-            if state.lower() == "up":
-                # Get the interface speed - used to select the fastest available
-                with open(os.path.join(net_path, device, "speed"), "r") as \
+    if not args.list:
+        # Get the default interface to use if OFI_INTERFACE is not set
+        interface = os.environ.get("OFI_INTERFACE")
+        if interface is None:
+            # Find all the /sys/class/net interfaces on the launch node
+            # (excluding lo)
+            print("Detecting network devices - OFI_INTERFACE not set")
+            available_interfaces = {}
+            net_path = os.path.join(os.path.sep, "sys", "class", "net")
+            net_list = [dev for dev in os.listdir(net_path) if dev != "lo"]
+            for device in sorted(net_list):
+                if device == "bonding_masters":
+                    continue
+                # Get the interface state - only include active (up) interfaces
+                with open(os.path.join(net_path, device, "operstate"), "r") as \
                      fileh:
-                    try:
-                        speed = int(fileh.read().strip())
-                        # KVM/Qemu/libvirt returns an EINVAL
-                    except IOError as ioerror:
-                        if ioerror.errno == errno.EINVAL:
-                            speed = 1000
-                        else:
-                            raise
+                    state = fileh.read().strip()
+                # Only include interfaces that are up
+                if state.lower() == "up":
+                    # Get the interface speed - used to select the fastest
+                    # available
+                    with open(os.path.join(net_path, device, "speed"), "r") as \
+                         fileh:
+                        try:
+                            speed = int(fileh.read().strip())
+                            # KVM/Qemu/libvirt returns an EINVAL
+                        except IOError as ioerror:
+                            if ioerror.errno == errno.EINVAL:
+                                speed = 1000
+                            else:
+                                raise
+                    print(
+                        "  - {0:<5} (speed: {1:>6} state: {2})".format(
+                            device, speed, state))
+                    # Only include the first active interface for each speed
+                    # First # is determined by an alphabetic sort: ib0 will be
+                    # checked before ib1
```

Comment:
Small correction on the updated indent:
```suggestion
                    # Only include the first active interface for each speed -
                    # first is determined by an alphabetic sort: ib0 will be
                    # checked before ib1
```